src/fty_asset_server.cc

```
@@ -234,19 +233,12 @@ fty_asset_server_new (void)
     fty_asset_server_t *self = (fty_asset_server_t *) zmalloc (sizeof (fty_asset_server_t));
     assert (self);
     self->mailbox_client = mlm_client_new ();
-    if (self->mailbox_client) {
-        self->verbose = false;
-    }
-    else {
+    if (!self->mailbox_client)
         fty_asset_server_destroy (&self);
```

It would make sense to me to add exit(1) or something like that, in the rare case when mlm_client_new fails, we will get segfault instead of error log and reasonable exit

---

Definitely. But that's a topic for a separate PR I guess.